includes/class-wc-gateway-stripe.php

```diff
@@ -532,17 +532,16 @@ public function maybe_remove_non_existent_customer( $error, $order ) {
 	 * @return array                      Redirection data for `process_payment`.
 	 */
 	public function complete_free_order( $order, $prepared_source, $force_save_source ) {
-		$response = array(
-			'result'   => 'success',
-			'redirect' => $this->get_return_url( $order ),
-		);
-
 		if ( $force_save_source ) {
 			$intent_secret = $this->setup_intent( $order, $prepared_source );
 
 			if ( ! empty( $intent_secret ) ) {
-				$response['setup_intent_secret'] = $intent_secret;
-				return $response;
+				// `get_return_url()` must be called immediately before returning a value.
```

Comment:
What if force_save_source is not true? Will moving the call to get_return_url inside this conditional be a problem?